graph-test/src/test/scala/net/spals/dsa/graph/WeightedGraphSpec.scala

```diff
@@ -1,18 +1,54 @@
 package net.spals.dsa.graph
 
-import org.scalatest.{FunSpec, Matchers}
+import org.scalatest.{GivenWhenThen, FunSpec, Matchers}
 
 /**
   * BDD tests for [[WeightedGraph]]
   *
   * @author tkral
   */
-class WeightedGraphSpec extends FunSpec with Matchers {
+class WeightedGraphSpec extends FunSpec with GivenWhenThen with Matchers {
 
   describe("A weighted graph") {
-    it("should have an empty state") {
+    it("should have an empty state without vertices") {
+      Given("an empty graph")
       val emptyGraph = WeightedGraph.empty
+
+      Then("no vertices should be present")
       emptyGraph.vertices.size should be (0)
     }
   }
+
+  it("should allow for vertices with edges") {
```

Comment:
Does it have an idea of parameterize testing?

like if you wanted to generalize a, b weight 1 and x, y weight 5